lib/gauge_loop_trace.cu

```diff
@@ -0,0 +1,74 @@
+#include <gauge_field.h>
+#include <gauge_force_quda.h>
+#include <instantiate.h>
+#include <tunable_reduction.h>
+#include <kernels/gauge_loop_trace.cuh>
+
+namespace quda {
+
+  template<typename Float, int nColor, QudaReconstructType recon>
+  class GaugeLoopTrace : public TunableMultiReduction {
+    const GaugeField &u;
+    using reduce_t = array<double, 2>;
+    std::vector<reduce_t>& loop_traces;
+    double factor;
+    const paths<1> p;
+
+  public:
+    // max block size of 8 is arbitrary for now
+    GaugeLoopTrace(const GaugeField &u, std::vector<reduce_t> &loop_traces, double factor, const paths<1>& p) :
+      TunableMultiReduction(u, 2u, p.num_paths, 8),
+      u(u),
+      loop_traces(loop_traces),
+      factor(factor),
+      p(p)
+    {
+      if (p.num_paths != loop_traces.size()) errorQuda("Loop traces size %lu != number of paths %d", loop_traces.size(), p.num_paths);
```

Comment:
Noting it's this line that is causing clang builds to fail (signed vs unsigned comparison)

Reply:
Done locally, will push momentarily

Reply:
https://github.com/lattice/quda/commit/4a893d83c4c5b4c15c9e0d07f8cc3cd1a0e1955a